request.go

```diff
@@ -124,3 +128,40 @@ func detectEncoding(opts RequestOptions) encoding.Encoding {
 
 	return encoding.JSON
 }
+
+// prepares the request by injecting metadata, applying plugin-based transport middleware,
+// before finally adding any user-provided override values
+func prepareRequest(req *transport.Request, headers map[string]string, opts Options) (*transport.Request, error) {
+	// Add request metadata
+	req.TargetService = opts.TOpts.ServiceName
```

Comment:
the fact that service is set, but headers are not seems like a strange design. we should consistently set everything in the same place